doc/develop/application/index.rst

```diff
@@ -474,7 +474,20 @@ Application Configuration Directory
 Zephyr will use configuration files from the application's configuration
 directory except for files with an absolute path provided by the arguments
 described earlier, for example ``CONF_FILE``, ``OVERLAY_CONFIG``, and
-``DTC_OVERLAY_FILE``.
+``DTC_OVERLAY_FILE``.  The absolute path to a file in a module will be
+generated by referring to it in your project's :file:`CMakeLists.txt` as:
+
+.. code-block:: cmake
+
+   set(CONF_FILE "\${ZEPHYR_MODULENAME_MODULE_DIR}/<module-relative-pathname>")
```

Comment:
Could you please push the correct version ?

This is old version where changes was requested here:
https://github.com/zephyrproject-rtos/zephyr/pull/42057#discussion_r807340997

**note** you most likely need to manually expand collapsed comments in githug for the link to take you to the right comment.
Old comment copy:
> Looking at the rendered output, I feel this particular feature fills to much, compared to the description of the other parts:
![image](https://user-images.githubusercontent.com/9802583/154154022-eb50bd05-19f4-4282-83be-81c4b4ca8b09.png)
>
> How about just a short mentioning of this possibility, for example:
```suggestion
``DTC_OVERLAY_FILE``. A file in a Zephyr module can be referred by escaping the Zephyr module dir variable like this ``\${ZEPHYR_<module>_MODULE_DIR}/<path-to>/<file>`` when setting any of said variables.
```